### ISSUE TITLE
Proposal Peer Review by Team MildCats for Milestone 3

### ISSUE THREAD

**Issue by sethisankalp** [INFO-526-F24/project-01-Vizzards#3]
The following is the peer review of the project proposal by Team MildCats. The team members that participated in this review are 

- Prajwal Sathyanarayana - @PRAJWAL2108
- Sankalp Sethi - @sethisankalp
- Tirth Shah- @tirthshah1-DA
- Yash Salokhe - @Yash-Salokhe
- Siva Ravula - @Siva-Bhargav
- Vishnu Rendla - @visren
- Solman Sarva - @sarvasolomonraju

**Describe the goal of the project.**

- The project aims to analyze and profile various types of cyber attacks in IoT environments using network data (packet traffic flow), and formulate distinctive patterns in the network data that can be used to flag suspicious behaviour or potential attacks.

**Describe the data used or collected.**

- The RT-IoT2022 dataset used here has been sourced from the UCI Machine Learning repository. This is a simulated dataset and the data was captured from real time IoT devices by having the attacker and victim (IoT device) on the same network and using Wireshark to perform packet capture.
- The dataset contains network traffic flow data like network identifiers, protocol and service information, payload information and various other metadata.

**Describe how the research question will be answered, e.g. what approaches / methods will be used.**

- While the overall approach - "Analyzing the network flow data to identify characteristic behaviours during attacks" is reflected in the plan, the exact approach is not immediately clear
- As far as we can understand, various data points will be compared and benchmarked (state of data during an attack vs normal) to identify patterns that can indicate an attack

**Is there anything that is unclear from the proposal?**

- The proposal is highly technical and hard to follow without prior knowledge about pentesting, cybersecurity, packet capturing protocols etc. 
- A large number of columns are unexplained, while a description for each column is not expected, a general idea about what they represent and why they are relevant would be great
- The Analysis plan only describes the "What" and not the "Why" of the approach, e.g - For Question 1 - "Variable proto , service and id.resp_p will be used...." - Why these variables are relevant is not explained 
- While we do not expect actual visualizations or charts at this stage, some proposed chart types or how some of the information will be visualized would have helped us understand the project better.

**Provide constructive feedback on how the team might be able to improve their project.**

- A section or a flow-diagram (in the proposal) to build context around how packets flow and how capture works and is relevant to the questions asked in the proposal would be great - Since this is a highly technical topic, it would help simplify things for the viewer.  
- The research questions currently contain column names without a clear description of their relevance to the question e.g. (fwd_pkts_tot, bwd_pkts_tot, fwd_data_pkts_tot) - This can be explained briefly to make sure the proposal is clearer.
- The analysis plan can be expanded to include why the selected variables are relevant, and what we "expect" to see in our initial Analysis

**What aspect of this project are you most interested in and would like to see highlighted in the presentation.**

- We are interested to see whether the analysis can reveal characteristics of a "digital fingerprint" that can be used to identify incoming attacks based on the network flow data alone. 
- We are also interested to see what kind of visualizations will be used to represent the dense network flow data.

**Provide constructive feedback on any issues with file and/or code organization.**

- No issues with the file/code organization. The repository is clean, well organized and does not contain junk files.

**(Optional) Any further comments or feedback?**

- No further comments. Excited to see what the analysis reveals!

**Comment by Ash-Y-28** [comment]
Your feedback has been carefully considered and the team has taken steps to address the points you raised. We appreciate your insights and are working diligently to enhance the clarity and accessibility of our proposal. Thank you for your constructive comments and thank you for your support!
